src/main/java/com/exasol/adapter/dynamodb/DynamodbAdapter.java

```diff
@@ -155,15 +156,21 @@ private String runQuery(final ExaMetadata exaMetadata, final PushDownRequest req
     }
 
     /**
-     * Get the total number of cores that are available in the cluster. This method assumes that all cluster nodes have
-     * an equal number of cores.
+     * Get the number of cores that can be used by a query. This methods calculates the number of available cores and
+     * limits it by the configured allowed amount.
+     * 
+     * @implNote This method assumes that all cluster nodes have * an equal number of cores.
      *
-     * @param exaMetadata {@link ExaMetadata}
+     * @param exaMetadata       {@link ExaMetadata}
+     * @param adapterProperties adapter properties
      * @return total number of cores that are available in the cluster
      */
-    private int getAvailableClusterCores(final ExaMetadata exaMetadata) {
+    private int getMaxCoreNumber(final ExaMetadata exaMetadata, final AdapterProperties adapterProperties)
+            throws AdapterException {
         final int cores = Runtime.getRuntime().availableProcessors();
-        return (int) (exaMetadata.getNodeCount() * cores);
+        final DynamodbAdapterProperties dynamodbAdapterProperties = new DynamodbAdapterProperties(adapterProperties);
+        final int maxConfiguredCores = dynamodbAdapterProperties.getMaxParallelUdfs();
+        return (int) Math.min((exaMetadata.getNodeCount() * cores), maxConfiguredCores);
```

Comment:
Do you need cast here? I think Math.min() returns int

Reply:
i mopved it inside the min